daffodil-core/src/main/scala/org/apache/daffodil/dsom/Term.scala

```diff
@@ -81,6 +82,51 @@ trait HasTermCheck {
   }
 }
 
+/**
+ * A set of possibilities for elements that can arrive as events
+ * in a valid infoset.
```

Comment:
Will rename "Sibling" to "PossibleNextElement", or perhaps PNE because these are case classes and tests are going to pattern match on these quite a bit. 

I can rename isRequiredInInfoset to match isRequiredStreamingUnparserEvent, but perhaps it should be overrideIsRequiredStreamingUnparserEvent.